### PR TITLE
Review changes

### PR DESCRIPTION
**Pull request by finestructure**
Hey @daveverwer these are the things we discussed yesterday. I'll go ahead and merge this now into the ongoing `populate-versions` branch - just raising this as an easy way to FYI.

### CODE REVIEW

File: Sources/App/Commands/Analyzer.swift [hunk 1]
```diff
@@ -3,26 +3,26 @@ import Vapor
 import ShellOut
 
 
-struct InspectorCommand: Command {
+struct AnalyzerCommand: Command {
```
Comment (finestructure) [inline Sources/App/Commands/Analyzer.swift:6]: I went with Anaylzer for now to sort of indicate that it's a "heavier" operation. Let me know if you've got better ideas!

File: Sources/App/Commands/Ingestor.swift [hunk 2]
```diff
@@ -26,40 +26,43 @@ struct IngestorCommand: Command {
...
-                        pkg.status = .ok
-                        return pkg.save(on: database)
-                    }
-            } catch {
-                return recordIngestionError(database: database, error: error)
-            }
-        }
+    Package.fetchUpdateCandidates(database, limit: limit)
+        .flatMapEach(on: database.eventLoop) { fetchMetadata(for: $0, with: client) }
+        .flatMapEachThrowing { updateTables(on: database, result: $0) }
```
Comment (finestructure) [inline Sources/App/Commands/Ingestor.swift:31]: As discussed, I pulled metadata fetching out into the top level `ingest` command and also broke out the table updates into their own method. Reads nicer now, I think.

File: Sources/App/Controllers/PackageController.swift [hunk 3]
```diff
@@ -58,6 +63,7 @@ extension PackageController {
     enum Command: String {
         case reconcile
         case ingest
+        case analyze
```
Comment (finestructure) [inline Sources/App/Controllers/PackageController.swift:66]: This is the API hook to trigger analysis that I added during our walk-through.

File: Sources/App/Models/Package.swift [hunk 4]
```diff
@@ -96,11 +96,12 @@ extension QueryBuilder where Model == Package {
...
 
 
-extension QueryBuilder where Model == Package {
-    func updateCandidates(limit: Int) -> EventLoopFuture<[Package]> {
-        // TODO: filter out updated in last X minutes
-        sort(\.$updatedAt)
-        .limit(limit)
-        .all()
+extension Package {
+    static func fetchUpdateCandidates(_ database: Database, limit: Int) -> EventLoopFuture<[Package]> {
```
Comment (finestructure) [inline Sources/App/Models/Package.swift:100]: Renamed as discussed

File: Sources/App/Commands/Analyzer.swift (hunk 1 above)
Comment (daveverwer) [reply Sources/App/Commands/Analyzer.swift:6]: There's no ideal names, but I think having one start with an `A` and one with an `I` will at least make it easier to keep a mental model of which does what 😂

**Comment by daveverwer** [comment]
Fab!

**Pull request merged**